src/Model/Scope/CompoundCondition.php

```diff
@@ -41,22 +41,22 @@ public function __construct(array $nestedConditions = [], string $junction = sel
         $this->junction = $junction;
 
         foreach ($nestedConditions as $nestedCondition) {
-            $nestedCondition = is_string($nestedCondition) ? new Condition($nestedCondition) : $nestedCondition;
+            if ($nestedCondition instanceof AbstractCondition) {
+                $condition = $nestedCondition;
+            } else {
+                if (!is_array($nestedCondition)) {
```

Comment:
maybe use one-liner casting 
`$nestedCondition = (array) $nestedCondition;`

Reply:
no! we assigning the same is a bad design, really